tests/components/apsystems/test_config_flow.py

```diff
@@ -57,6 +75,33 @@ async def test_form_cannot_connect_and_recover(
     assert result2["data"].get(CONF_IP_ADDRESS) == "127.0.0.1"
 
 
+async def test_form_cannot_connect_and_recover_custom_port(
+    hass: HomeAssistant, mock_apsystems: AsyncMock, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we handle cannot connect error but recovering with custom port."""
+
+    mock_apsystems.get_device_info.side_effect = TimeoutError
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_USER},
+        data={CONF_IP_ADDRESS: "127.0.0.2", CONF_PORT: 8042},
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    mock_apsystems.get_device_info.side_effect = None
+
+    result2 = await hass.config_entries.flow.async_configure(
+        result["flow_id"],
+        {CONF_IP_ADDRESS: "127.0.0.1", CONF_PORT: 8042},
```

Comment:
Idk how important that is, but maybe set the 2nd attempt to another port, so we can see that it accepts changes after crashes.

Reply:
We are not testing library here. So that type of test should go into lib